src/microbe_stage/AgentProjectile.cs

```diff
@@ -21,6 +25,13 @@ public override void _Ready()
     {
         AddCollisionExceptionWith(Emitter);
         Connect("body_entered", this, "OnBodyEntered");
+
+        // Timer that delay despawn of projectiles
+        DespawnTimer = new Timer();
+        DespawnTimer.OneShot = true;
+        DespawnTimer.WaitTime = Delaydespawn;
+        DespawnTimer.Connect("timeout", this, "OnTimerTimeout");
+        AddChild(DespawnTimer);
```

Comment:
In order to support saving, this is not a very good approach. Instead a few new properties should be added like:
```
[JSONProperty]
bool FadingOut;
[JSONProperty]
float FadeTimeRemaining;
bool PhysicsDestroyed;
```

Then in _Process the time should be reduced, this destroyed when it reaches zero. And finally if PhysicsDestroyed is not true, then physics should be disabled and the velocity of this killed. That way saving should work without any complications.